src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php

```diff
@@ -282,7 +282,7 @@ private function round($number)
                     break;
             }
 
-            $number /= $roundingCoef;
+            $number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
```

Comment:
Can you add some tests please? Do you need help to write them?

Reply:
Done, it was easy to add.

$this-> assertSame fail without the following modification. (Failed asserting that -1234.0 is identical to -1234.)
```
$number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
```

Reply:
What about changing the `if` above?

```php
if (null !== $this->scale && 0 !== $scale && null !== $this->roundingMode) {
   // ...
}
```

Reply:
@xabbuh  If the user submit the following value `1,23` with the scale `null`, I expect to save the value `1,23`.
If the user submit the following value `1,23` with the scale `0`, I expect to save the rounded value `1`.

If I make your proposed change, i will save the value `1,23` in both cases.